R/sql-build.R

```diff
@@ -60,46 +66,49 @@ sql_render <- function(query,
 sql_render.tbl_lazy <- function(query,
                                 con = query$src$con,
                                 ...,
-                                use_star = TRUE,
                                 subquery = FALSE,
-                                lvl = 0,
-                                cte = FALSE) {
+                                lvl = 0) {
   sql_render(
     query$lazy_query,
     con = con,
     ...,
-    use_star = use_star,
     subquery = subquery,
-    lvl = lvl,
-    cte = cte
+    lvl = lvl
   )
 }
 
 #' @export
 sql_render.lazy_query <- function(query,
                                   con = NULL,
                                   ...,
-                                  use_star = TRUE,
                                   subquery = FALSE,
-                                  lvl = 0,
-                                  cte = FALSE) {
-  qry <- sql_build(query, con = con, ..., use_star = use_star)
-  qry <- sql_optimise(qry, con = con, ..., subquery = subquery)
+                                  lvl = 0) {
+  # FIXME the handling of `...` seems wrong overall.
```

Comment:
I didn't want to add `sql_options` to `sql_render()` as the options aren't needed in any other `sql_render()` method but only the `sql_build()` methods. This feels like a hack overall but is due to the fact that the SQL generation works like:

1. Call `sql_render()` on <tbl_lazy> resp. <lazy_query>
2. Build the lazy query via `sql_build()` (here the `sql_options` are needed; in there `sql_render()` is never called).
3. Optimise the resulting query via `sql_optimise()`.
4. Render the query via `sql_render()` (here the `sql_options` aren't needed).

Reply:
Even given that, I think you're still better off having an explicit `sql_options` argument rather than fishing out of `...`